### DISCUSSION TITLE
SSLHandshakeException running under Docker

### DISCUSSION THREAD

**Discussion by sproket** [nextgenhealthcare/connect#5456] [Q&A]
Anyone have this issue when running under Docker?   

javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

This occurs when mirth sends a web request. 

I assume the keystore file needs to be somewhere that can be read by the built-in Java? OR is it that it can't see the entry because the docker would have a different name than the host? 

from my docker file: 

FROM openjdk:11-jre

ENV MIRTH_CONNECT_VERSION 3.12.0.b2650

TIA

**Comment by jonbartels** [comment]
> This occurs when mirth sends a web request.

Sends a web request to where?

**Comment by sproket** [comment]
I post to another server to check login

URL: https://other_server/admin/user/system-login
METHOD: POST
etc...

**Reply by jonbartels** [reply]
Does the other server present a certificate signed by a root CA or a self-signed or internally signed cert?

Are you doing one-way HTTPS or two-way mutual TLS?

Notes - https://gist.github.com/jonbartels/8abd121901eb930f46245d9ef0f5710e

**Reply by sproket** [reply]
One way to the other server.  Thanks. I'll check that.

**Comment by sproket** [comment]
I tried adding the keystore to the mcserver.vmoptions and mcservice.vmoptions. same problem:

-server
-Xmx256m
-Djava.awt.headless=true
-Dapple.awt.UIElement=true
-Djavax.net.ssl.trustStore=/opt/mirth-connect/appdata/keystore.jks
